src/Microsoft.AspNetCore.Diagnostics.Elm/ElmExtensions.cs

```diff
@@ -22,9 +21,12 @@ public static IApplicationBuilder UseElmCapture(this IApplicationBuilder app)
             }
 
             // add the elm provider to the factory here so the logger can start capturing logs immediately
-            var factory = app.ApplicationServices.GetRequiredService<ILoggerFactory>();
-            var provider = app.ApplicationServices.GetRequiredService<ElmLoggerProvider>();
-            factory.AddProvider(provider);
+            var factory = app.ApplicationServices.GetRequiredService<ILoggerFactory>() as LoggerFactory;
```

Comment:
Because `ILoggerFactory` is obsoleting `AddProvider` it's now just a method on `LoggerFactory`